src/components/Contacts.js

```diff
@@ -0,0 +1,161 @@
+import React, { Fragment } from 'react';
+import { compose, withState, withHandlers } from 'recompose';
+
+// global
+import * as GC from '../constants';
+// components
+import { SectionTitle } from './SectionTitle';
+import { ContactButtons } from './ContactButtons';
+// helpers
+import * as H from '../helpers';
+// ui
+import {
+  FormInput,
+  FormTextArea,
+  ContactsInfo,
+  FormContainer,
+  ContactSection,
+  FormFieldContainer,
+} from './ui';
+//  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+const fields = [
+  {
+    type: 'text',
+    name: 'yourName',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'email',
+    name: 'email',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'tel',
+    name: 'phoneNumber',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'text',
+    name: 'details',
+    component: (props) => <FormTextArea {...props} />,
+  },
+];
+
+const initialFieldValues = {
+  yourName: '',
+  email: '',
+  phoneNumber: '',
+  details: '',
+};
+
+const initialErrors = {
+  yourName: false,
+  email: false,
+  phoneNumber: false,
+  details: false,
+};
+
+const getErrors = (values) => ({
+  yourName: !GC.USER_NAME_REGEX.test(values.yourName),
+  email: !GC.EMAIL_REGEX.test(values.email),
+  phoneNumber: !GC.PHONE_NUMBER_REGEX.test(values.phoneNumber),
+  details: false,
+});
+
+const enhance = compose(
+  withState('fields', 'setFields', { ...initialFieldValues }),
+  withState('errors', 'setErrors', { ...initialErrors }),
+  withState('requestPending', 'setRequestPending', false),
+  withHandlers({
+    handleHideError: (props) => (name) => {
+      const newErrors = props.errors;
+      newErrors[name] = false;
+      props.setErrors(newErrors);
+    },
+    handleFieldChange: (props) => (e, name) => {
+      const field = e.currentTarget;
+      const newFields = props.fields;
+      newFields[name] = field.value;
+      props.setFields(newFields);
+    },
+    handleSubmit: (props) => () => {
+      const errors = getErrors(props.fields);
+      const isInvalid = Object.values(errors).some(err => err === true);
```

Comment:
isInvalid not correct name for boolean. use just `invalid`. Name with `is` must be a function

Reply:
TODO: check everywhere isInvalid and change to invalid